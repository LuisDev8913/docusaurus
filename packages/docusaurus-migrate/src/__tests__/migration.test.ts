/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {migrateDocusaurusProject} from '../index';
import path from 'path';
import fs from 'fs-extra';

describe('migration test', () => {
  test('simple website', async () => {
    const siteDir = path.join(
      __dirname,
      '__fixtures__',
      'simple_website',
      'website',
    );
    const newDir = path.join(__dirname, '__fixtures__', 'migrated_simple_site');
    await expect(
      migrateDocusaurusProject(siteDir, newDir),
    ).resolves.toBeUndefined();
    await fs.remove(newDir);
  });
  test('complex website', async () => {
    const siteDir = path.join(
      __dirname,
      '__fixtures__',
      'complex_website',
      'website',
    );
    const newDir = path.join(
      __dirname,
      '__fixtures__',
      'migrated_complex_site',
    );
    await expect(
      migrateDocusaurusProject(siteDir, newDir),
    ).resolves.toBeUndefined();
    await fs.remove(newDir);
  });

  test('missing versions', async () => {
    const siteDir = path.join(
      __dirname,
      '__fixtures__',
      'missing_version_website',
      'website',
    );
    const newDir = path.join(
      __dirname,
      '__fixtures__',
      'migrated_missing_version_site',
    );
    await expect(
      migrateDocusaurusProject(siteDir, newDir),
    ).resolves.toBeUndefined();
    await fs.remove(newDir);
  });
});
