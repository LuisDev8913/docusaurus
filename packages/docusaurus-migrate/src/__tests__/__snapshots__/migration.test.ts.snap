// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`migration test complex website: copy 1`] = `
Array [
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/complex_website/website/static",
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_complex_site/static",
  ],
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/complex_website/website/blog",
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_complex_site/blog",
  ],
]
`;

exports[`migration test complex website: mkdirp 1`] = `
Array [
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_complex_site/src/pages/",
  ],
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_complex_site/src/css",
  ],
]
`;

exports[`migration test complex website: mkdirs 1`] = `Array []`;

exports[`migration test complex website: write 1`] = `
Array [
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_complex_site/src/pages/index.js",
    "import Layout from \\"@theme/Layout\\";
      import React from \\"react\\";

      export default () => {
        return <Layout />;
      };
      ",
  ],
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_complex_site/src/css/customTheme.css",
    ":root{
  --ifm-color-primary-lightest: #3CAD6E;
  --ifm-color-primary-lighter: #359962;
  --ifm-color-primary-light: #33925D;
  --ifm-color-primary: #2E8555;
  --ifm-color-primary-dark: #29784C;
  --ifm-color-primary-darker: #277148;
  --ifm-color-primary-darkest: #205D3B;
}
",
  ],
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_complex_site/docusaurus.config.js",
    "module.exports={
  \\"title\\": \\"Docusaurus\\",
  \\"tagline\\": \\"Easy to Maintain Open Source Documentation Websites\\",
  \\"url\\": \\"https://docusaurus.io\\",
  \\"baseUrl\\": \\"/\\",
  \\"organizationName\\": \\"facebook\\",
  \\"projectName\\": \\"docusaurus\\",
  \\"noIndex\\": false,
  \\"scripts\\": [
    \\"https://buttons.github.io/buttons.js\\",
    \\"https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.0/clipboard.min.js\\",
    \\"/js/code-blocks-buttons.js\\"
  ],
  \\"favicon\\": \\"img/docusaurus.ico\\",
  \\"customFields\\": {
    \\"users\\": {
      \\"caption\\": \\"DevSpace\\",
      \\"image\\": \\"/img/users/devspace.svg\\",
      \\"infoLink\\": \\"https://devspace.cloud/docs/\\",
      \\"fbOpenSource\\": false,
      \\"pinned\\": false
    },
    \\"translationRecruitingLink\\": \\"https://crowdin.com/project/docusaurus\\",
    \\"facebookAppId\\": \\"199138890728411\\"
  },
  \\"onBrokenLinks\\": \\"log\\",
  \\"onBrokenMarkdownLinks\\": \\"log\\",
  \\"presets\\": [
    [
      \\"@docusaurus/preset-classic\\",
      {
        \\"docs\\": {
          \\"showLastUpdateAuthor\\": true,
          \\"showLastUpdateTime\\": true,
          \\"editUrl\\": \\"https://github.com/facebook/docusaurus/edit/main/docs/\\"
        },
        \\"blog\\": {},
        \\"theme\\": {
          \\"customCss\\": \\"../complex_website/src/css/customTheme.css\\"
        },
        \\"googleAnalytics\\": {
          \\"trackingID\\": \\"UA-44373548-31\\"
        }
      }
    ]
  ],
  \\"plugins\\": [],
  \\"themeConfig\\": {
    \\"navbar\\": {
      \\"title\\": \\"Docusaurus\\",
      \\"logo\\": {
        \\"src\\": \\"img/docusaurus.svg\\"
      },
      \\"items\\": [
        {
          \\"to\\": \\"docs/installation\\",
          \\"label\\": \\"Docs\\",
          \\"position\\": \\"left\\"
        },
        {
          \\"to\\": \\"docs/tutorial-setup\\",
          \\"label\\": \\"Tutorial\\",
          \\"position\\": \\"left\\"
        },
        {
          \\"to\\": \\"/users\\",
          \\"label\\": \\"Users\\",
          \\"position\\": \\"left\\"
        },
        {
          \\"href\\": \\"https://github.com/facebook/docusaurus\\",
          \\"label\\": \\"GitHub\\",
          \\"position\\": \\"left\\"
        }
      ]
    },
    \\"image\\": \\"img/docusaurus.png\\",
    \\"footer\\": {
      \\"links\\": [
        {
          \\"title\\": \\"Community\\",
          \\"items\\": [
            {
              \\"label\\": \\"Twitter\\",
              \\"to\\": \\"https://twitter.com/docusaurus\\"
            }
          ]
        }
      ],
      \\"copyright\\": \\"Copyright Â© 2022 Facebook Inc.\\",
      \\"logo\\": {
        \\"src\\": \\"img/docusaurus_monochrome.svg\\"
      }
    },
    \\"algolia\\": {
      \\"apiKey\\": \\"3eb9507824b8be89e7a199ecaa1a9d2c\\",
      \\"indexName\\": \\"docusaurus\\",
      \\"algoliaOptions\\": {
        \\"facetFilters\\": [
          \\"language:LANGUAGE\\",
          \\"version:VERSION\\"
        ]
      }
    }
  }
}",
  ],
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_complex_site/package.json",
    "{
  \\"name\\": \\"docusaurus-1-website\\",
  \\"version\\": \\"2.0.0-alpha.58\\",
  \\"private\\": true,
  \\"scripts\\": {
    \\"start\\": \\"docusaurus start\\",
    \\"build\\": \\"docusaurus build\\",
    \\"publish-gh-pages\\": \\"docusaurus-publish\\",
    \\"examples\\": \\"docusaurus-examples\\",
    \\"write-translations\\": \\"docusaurus-write-translations\\",
    \\"docusaurus-version\\": \\"docusaurus-version\\",
    \\"rename-version\\": \\"docusaurus-rename-version\\",
    \\"crowdin-upload\\": \\"crowdin --config ../crowdin.yaml upload sources --auto-update -b master\\",
    \\"crowdin-download\\": \\"crowdin --config ../crowdin.yaml download -b master\\",
    \\"swizzle\\": \\"docusaurus swizzle\\",
    \\"deploy\\": \\"docusaurus deploy\\",
    \\"docusaurus\\": \\"docusaurus\\"
  },
  \\"dependencies\\": {
    \\"@docusaurus/core\\": \\"2.0.0-beta.15\\",
    \\"@docusaurus/preset-classic\\": \\"2.0.0-beta.15\\",
    \\"clsx\\": \\"^1.1.1\\",
    \\"react\\": \\"^17.0.2\\",
    \\"react-dom\\": \\"^17.0.2\\"
  }
}",
  ],
]
`;

exports[`migration test missing versions: copy 1`] = `
Array [
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/missing_version_website/website/static",
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_missing_version_site/static",
  ],
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/missing_version_website/website/blog",
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_missing_version_site/blog",
  ],
]
`;

exports[`migration test missing versions: mkdirp 1`] = `
Array [
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_missing_version_site/src/pages/",
  ],
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_missing_version_site/src/css",
  ],
]
`;

exports[`migration test missing versions: mkdirs 1`] = `Array []`;

exports[`migration test missing versions: write 1`] = `
Array [
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_missing_version_site/src/pages/index.js",
    "import Layout from \\"@theme/Layout\\";
      import React from \\"react\\";

      export default () => {
        return <Layout />;
      };
      ",
  ],
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_missing_version_site/src/css/customTheme.css",
    ":root{
  --ifm-color-primary-lightest: #3CAD6E;
  --ifm-color-primary-lighter: #359962;
  --ifm-color-primary-light: #33925D;
  --ifm-color-primary: #2E8555;
  --ifm-color-primary-dark: #29784C;
  --ifm-color-primary-darker: #277148;
  --ifm-color-primary-darkest: #205D3B;
}
",
  ],
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_missing_version_site/docusaurus.config.js",
    "module.exports={
  \\"title\\": \\"Docusaurus\\",
  \\"tagline\\": \\"Easy to Maintain Open Source Documentation Websites\\",
  \\"url\\": \\"https://docusaurus.io\\",
  \\"baseUrl\\": \\"/\\",
  \\"organizationName\\": \\"facebook\\",
  \\"projectName\\": \\"docusaurus\\",
  \\"noIndex\\": false,
  \\"scripts\\": [
    \\"https://buttons.github.io/buttons.js\\",
    \\"https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.0/clipboard.min.js\\",
    \\"/js/code-blocks-buttons.js\\"
  ],
  \\"favicon\\": \\"img/docusaurus.ico\\",
  \\"customFields\\": {
    \\"users\\": {
      \\"caption\\": \\"DevSpace\\",
      \\"image\\": \\"/img/users/devspace.svg\\",
      \\"infoLink\\": \\"https://devspace.cloud/docs/\\",
      \\"fbOpenSource\\": false,
      \\"pinned\\": false
    },
    \\"translationRecruitingLink\\": \\"https://crowdin.com/project/docusaurus\\",
    \\"facebookAppId\\": \\"199138890728411\\"
  },
  \\"onBrokenLinks\\": \\"log\\",
  \\"onBrokenMarkdownLinks\\": \\"log\\",
  \\"presets\\": [
    [
      \\"@docusaurus/preset-classic\\",
      {
        \\"docs\\": {
          \\"showLastUpdateAuthor\\": true,
          \\"showLastUpdateTime\\": true,
          \\"editUrl\\": \\"https://github.com/facebook/docusaurus/edit/main/docs/\\"
        },
        \\"blog\\": {},
        \\"theme\\": {
          \\"customCss\\": \\"../missing_version_website/src/css/customTheme.css\\"
        },
        \\"googleAnalytics\\": {
          \\"trackingID\\": \\"UA-44373548-31\\"
        }
      }
    ]
  ],
  \\"plugins\\": [],
  \\"themeConfig\\": {
    \\"navbar\\": {
      \\"title\\": \\"Docusaurus\\",
      \\"logo\\": {
        \\"src\\": \\"img/docusaurus.svg\\"
      },
      \\"items\\": [
        {
          \\"to\\": \\"docs/installation\\",
          \\"label\\": \\"Docs\\",
          \\"position\\": \\"left\\"
        },
        {
          \\"to\\": \\"docs/tutorial-setup\\",
          \\"label\\": \\"Tutorial\\",
          \\"position\\": \\"left\\"
        },
        {
          \\"to\\": \\"/users\\",
          \\"label\\": \\"Users\\",
          \\"position\\": \\"left\\"
        },
        {
          \\"href\\": \\"https://github.com/facebook/docusaurus\\",
          \\"label\\": \\"GitHub\\",
          \\"position\\": \\"left\\"
        }
      ]
    },
    \\"image\\": \\"img/docusaurus.png\\",
    \\"footer\\": {
      \\"links\\": [
        {
          \\"title\\": \\"Community\\",
          \\"items\\": [
            {
              \\"label\\": \\"Twitter\\",
              \\"to\\": \\"https://twitter.com/docusaurus\\"
            }
          ]
        }
      ],
      \\"copyright\\": \\"Copyright Â© 2022 Facebook Inc.\\",
      \\"logo\\": {
        \\"src\\": \\"img/docusaurus_monochrome.svg\\"
      }
    },
    \\"algolia\\": {
      \\"apiKey\\": \\"3eb9507824b8be89e7a199ecaa1a9d2c\\",
      \\"indexName\\": \\"docusaurus\\",
      \\"algoliaOptions\\": {
        \\"facetFilters\\": [
          \\"language:LANGUAGE\\",
          \\"version:VERSION\\"
        ]
      }
    }
  }
}",
  ],
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_missing_version_site/package.json",
    "{
  \\"name\\": \\"docusaurus-1-website\\",
  \\"version\\": \\"2.0.0-alpha.58\\",
  \\"private\\": true,
  \\"scripts\\": {
    \\"start\\": \\"docusaurus start\\",
    \\"build\\": \\"docusaurus build\\",
    \\"publish-gh-pages\\": \\"docusaurus-publish\\",
    \\"examples\\": \\"docusaurus-examples\\",
    \\"write-translations\\": \\"docusaurus-write-translations\\",
    \\"docusaurus-version\\": \\"docusaurus-version\\",
    \\"rename-version\\": \\"docusaurus-rename-version\\",
    \\"crowdin-upload\\": \\"crowdin --config ../crowdin.yaml upload sources --auto-update -b master\\",
    \\"crowdin-download\\": \\"crowdin --config ../crowdin.yaml download -b master\\",
    \\"swizzle\\": \\"docusaurus swizzle\\",
    \\"deploy\\": \\"docusaurus deploy\\",
    \\"docusaurus\\": \\"docusaurus\\"
  },
  \\"dependencies\\": {
    \\"@docusaurus/core\\": \\"2.0.0-beta.15\\",
    \\"@docusaurus/preset-classic\\": \\"2.0.0-beta.15\\",
    \\"clsx\\": \\"^1.1.1\\",
    \\"react\\": \\"^17.0.2\\",
    \\"react-dom\\": \\"^17.0.2\\"
  }
}",
  ],
]
`;

exports[`migration test simple website: copy 1`] = `
Array [
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/simple_website/website/static",
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_simple_site/static",
  ],
]
`;

exports[`migration test simple website: mkdirp 1`] = `
Array [
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_simple_site/src/pages/",
  ],
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_simple_site/src/css",
  ],
]
`;

exports[`migration test simple website: mkdirs 1`] = `Array []`;

exports[`migration test simple website: write 1`] = `
Array [
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_simple_site/src/pages/index.js",
    "import Layout from \\"@theme/Layout\\";
      import React from \\"react\\";

      export default () => {
        return <Layout />;
      };
      ",
  ],
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/simple_website/docs/api-commands.md",
    "---
id: commands
title: CLI Commands
---

## Doc ",
  ],
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/simple_website/docs/api-doc-markdown.md",
    "---
id: doc-markdown
title: Markdown Features
---

## Doc",
  ],
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_simple_site/src/css/customTheme.css",
    ":root{
  --ifm-color-primary-lightest: #3CAD6E;
  --ifm-color-primary-lighter: #359962;
  --ifm-color-primary-light: #33925D;
  --ifm-color-primary: #2E8555;
  --ifm-color-primary-dark: #29784C;
  --ifm-color-primary-darker: #277148;
  --ifm-color-primary-darkest: #205D3B;
}
",
  ],
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_simple_site/docusaurus.config.js",
    "module.exports={
  \\"title\\": \\"Docusaurus\\",
  \\"tagline\\": \\"Easy to Maintain Open Source Documentation Websites\\",
  \\"url\\": \\"https://docusaurus.io\\",
  \\"baseUrl\\": \\"/\\",
  \\"organizationName\\": \\"facebook\\",
  \\"projectName\\": \\"docusaurus\\",
  \\"noIndex\\": false,
  \\"scripts\\": [
    \\"https://buttons.github.io/buttons.js\\",
    \\"https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.0/clipboard.min.js\\",
    \\"/js/code-blocks-buttons.js\\"
  ],
  \\"favicon\\": \\"img/docusaurus.ico\\",
  \\"customFields\\": {
    \\"users\\": {
      \\"caption\\": \\"DevSpace\\",
      \\"image\\": \\"/img/users/devspace.svg\\",
      \\"infoLink\\": \\"https://devspace.cloud/docs/\\",
      \\"fbOpenSource\\": false,
      \\"pinned\\": false
    },
    \\"translationRecruitingLink\\": \\"https://crowdin.com/project/docusaurus\\",
    \\"facebookAppId\\": \\"199138890728411\\"
  },
  \\"onBrokenLinks\\": \\"log\\",
  \\"onBrokenMarkdownLinks\\": \\"log\\",
  \\"presets\\": [
    [
      \\"@docusaurus/preset-classic\\",
      {
        \\"docs\\": {
          \\"showLastUpdateAuthor\\": true,
          \\"showLastUpdateTime\\": true,
          \\"editUrl\\": \\"https://github.com/facebook/docusaurus/edit/main/docs/\\",
          \\"path\\": \\"../simple_website/docs\\"
        },
        \\"blog\\": {},
        \\"theme\\": {
          \\"customCss\\": \\"../simple_website/src/css/customTheme.css\\"
        },
        \\"googleAnalytics\\": {
          \\"trackingID\\": \\"UA-44373548-31\\"
        }
      }
    ]
  ],
  \\"plugins\\": [],
  \\"themeConfig\\": {
    \\"navbar\\": {
      \\"title\\": \\"Docusaurus\\",
      \\"logo\\": {
        \\"src\\": \\"img/docusaurus.svg\\"
      },
      \\"items\\": [
        {
          \\"to\\": \\"docs/installation\\",
          \\"label\\": \\"Docs\\",
          \\"position\\": \\"left\\"
        },
        {
          \\"to\\": \\"docs/tutorial-setup\\",
          \\"label\\": \\"Tutorial\\",
          \\"position\\": \\"left\\"
        },
        {
          \\"to\\": \\"/users\\",
          \\"label\\": \\"Users\\",
          \\"position\\": \\"left\\"
        },
        {
          \\"href\\": \\"https://github.com/facebook/docusaurus\\",
          \\"label\\": \\"GitHub\\",
          \\"position\\": \\"left\\"
        }
      ]
    },
    \\"image\\": \\"img/docusaurus.png\\",
    \\"footer\\": {
      \\"links\\": [
        {
          \\"title\\": \\"Community\\",
          \\"items\\": [
            {
              \\"label\\": \\"Twitter\\",
              \\"to\\": \\"https://twitter.com/docusaurus\\"
            }
          ]
        }
      ],
      \\"copyright\\": \\"Copyright Â© 2022 Facebook Inc.\\",
      \\"logo\\": {
        \\"src\\": \\"img/docusaurus_monochrome.svg\\"
      }
    },
    \\"algolia\\": {
      \\"apiKey\\": \\"3eb9507824b8be89e7a199ecaa1a9d2c\\",
      \\"indexName\\": \\"docusaurus\\",
      \\"algoliaOptions\\": {
        \\"facetFilters\\": [
          \\"language:LANGUAGE\\",
          \\"version:VERSION\\"
        ]
      }
    }
  }
}",
  ],
  Array [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_simple_site/package.json",
    "{
  \\"name\\": \\"docusaurus-1-website\\",
  \\"version\\": \\"2.0.0-alpha.58\\",
  \\"private\\": true,
  \\"scripts\\": {
    \\"start\\": \\"docusaurus start\\",
    \\"build\\": \\"docusaurus build\\",
    \\"publish-gh-pages\\": \\"docusaurus-publish\\",
    \\"examples\\": \\"docusaurus-examples\\",
    \\"write-translations\\": \\"docusaurus-write-translations\\",
    \\"docusaurus-version\\": \\"docusaurus-version\\",
    \\"rename-version\\": \\"docusaurus-rename-version\\",
    \\"crowdin-upload\\": \\"crowdin --config ../crowdin.yaml upload sources --auto-update -b master\\",
    \\"crowdin-download\\": \\"crowdin --config ../crowdin.yaml download -b master\\",
    \\"swizzle\\": \\"docusaurus swizzle\\",
    \\"deploy\\": \\"docusaurus deploy\\",
    \\"docusaurus\\": \\"docusaurus\\"
  },
  \\"dependencies\\": {
    \\"@docusaurus/core\\": \\"2.0.0-beta.15\\",
    \\"@docusaurus/preset-classic\\": \\"2.0.0-beta.15\\",
    \\"clsx\\": \\"^1.1.1\\",
    \\"react\\": \\"^17.0.2\\",
    \\"react-dom\\": \\"^17.0.2\\"
  }
}",
  ],
]
`;
